### PR TITLE
added path support

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: path.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"fmt"
+	"strings"
+)
+
+type Path struct {
+	Nodes []interface{}
+	Edges []interface{}
+}
+
+func PathNew(nodes []interface{}, edges []interface{}) Path {
```
Comment (swilly22) [inline path.go]: `func PathNew(nodes []Node, edges []Edge) Path {`

File: path.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	return Path{
+		Nodes: nodes,
+		Edges: edges,
+	}
+}
+
+func (p Path) GetNodes() []interface{} {
```
Comment (swilly22) [inline path.go]: `func (p Path) GetNodes() []Node {`

File: path.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	return p.Nodes
+}
+
+func (p Path) GetEdges() []interface{} {
```
Comment (swilly22) [inline path.go]: `func (p Path) GetEdges() []Edge {`

File: path.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+func (p Path) GetNode(index int) *Node {
+	return p.Nodes[index].(*Node)
+}
+
+func (p Path) GetEdge(index int) *Edge{
+	return p.Edges[index].(*Edge);
+}
+
+func (p Path) FirstNode() *Node {
+	return p.GetNode(0)
```
Comment (swilly22) [inline path.go]: What if the Path is empty?

File: path.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+func (p Path) GetNodes() []interface{} {
+	return p.Nodes
+}
+
+func (p Path) GetEdges() []interface{} {
+	return p.Edges
+}
+
+func (p Path) GetNode(index int) *Node {
+	return p.Nodes[index].(*Node)
```
Comment (swilly22) [inline path.go]: Index validations?
index >=0 && index < Path length ?

File: path.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+func (p Path) GetEdge(index int) *Edge{
+	return p.Edges[index].(*Edge);
+}
+
+func (p Path) FirstNode() *Node {
+	return p.GetNode(0)
+}
+
+func (p Path) LastNode() *Node {
+	return p.GetNode(p.NodesCount() - 1)
```
Comment (swilly22) [inline path.go]: What if NodeCount is 0?

File: path.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+func (p Path) NodesCount() int {
+	return len(p.Nodes)
+}
+
+func (p Path) EdgeCount() int {
+	return len(p.Edges)
+}
+
+func (p Path) Encode() string {
```
Comment (swilly22) [inline path.go]: https://play.golang.org/p/Azql7_pDAA
`func (p Path) String() string {...`

File: README.md [hunk 8]
```diff
@@ -86,6 +86,14 @@ func main() {
...
         fmt.Printf("\nAge: %d\n", p_age)
     }
+
+    // Path matching example.
+    query = "MATCH p = (:Person)-[:Visited]->(:Country) RETURN p"
+    result, _ = graph.Query(query)
+    res.Next()
+    r := res.Record()
+    p, ok := r.GetByIndex(0).(Path)
+    fmt.Printf("%v", p.Encode())
```
Comment (swilly22) [inline README.md]: `fmt.Printf("%s", p)`

File: query_result.go [hunk 9]
```diff
@@ -217,6 +218,13 @@ func (qr *QueryResult) parseArray(cell interface{}) []interface{} {
 	return array
 }
 
+func (qr *QueryResult) parsePath(cell interface{}) Path {
+	arrays := cell.([]interface{})
+	nodes := qr.parseScalar(arrays[0].([]interface{}))
+	edges := qr.parseScalar(arrays[1].([]interface{}))
+	return PathNew(nodes.([]interface{}), edges.([]interface{}))
```
Comment (swilly22) [inline query_result.go]: `return PathNew(nodes, edges)`

File: path.go [hunk 10]
```diff
@@ -0,0 +1,69 @@
+package redisgraph
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Path struct {
+	Nodes []interface{}
+	Edges []interface{}
```
[the remark this answers is not in the document]
Comment (DvirDukhan) [reply path.go]: changed to `[]*Node` and `[]*Edge`

File: path.go (hunk 1 above)
Comment (DvirDukhan) [reply path.go]: the constructor will convert from array of interface to array of node/edge

File: path.go (hunk 3 above)
Comment (DvirDukhan) [reply path.go]: `[]*Edge`

File: path.go (hunk 4 above)
Comment (DvirDukhan) [reply path.go]: execption

File: path.go (hunk 5 above)
Comment (DvirDukhan) [reply path.go]: exception

File: path.go (hunk 6 above)
Comment (DvirDukhan) [reply path.go]: exception

File: query_result.go (hunk 9 above)
Comment (DvirDukhan) [reply query_result.go]: should be an array

File: README.md (hunk 8 above)
Comment (DvirDukhan) [reply README.md]: fixed

File: path.go (hunk 7 above)
Comment (DvirDukhan) [reply path.go]: fixed

**Pull request merged**